doc/application/index.rst

```diff
@@ -1535,17 +1535,23 @@ the C preprocessor on a file which includes the following:
 The Zephyr build system determines :makevar:`DTC_OVERLAY_FILE` as
 follows:
 
-- Any value given to :makevar:`DTC_OVERLAY_FILE` in your application
-  :file:`CMakeLists.txt` (**before including the boilerplate.cmake file**),
-  passed to the the CMake command line, or present in the CMake variable cache,
-  takes precedence.
-
-- The :ref:`environment variable <env_vars>` :envvar:`DTC_OVERLAY_FILE` is
-  checked next. This mechanism is now deprecated; users should set this
-  variable using CMake instead of the environment.
-
-- If the file :file:`BOARD.overlay` exists in your application directory,
-  where ``BOARD`` is the BOARD value set earlier, it will be used.
+1. Any value given to :makevar:`DTC_OVERLAY_FILE` in your application
```

Comment:
@dbkinder This was modeled after existing similar content for locating a configuration file in `Kconfig Configuration` earlier in the file.  Do you want me to change to this way only here, or to revise the other description as well as this?

Reply:
In answer to your question: in both cases the first source that is present is used; the remainder are ignored.

Reply:
Understood about the first source wins, but in your text it says CMakeLists.txt, command line, and CMake cache all in #1, so I wanted to clarify that.

Looking back at the BOARD value description above, I see that the cache actually has the highest precedence, then the command line, then an environment variable, and finally the CMakeLists.txt. Is that true for this DTC_OVERLAY_FILE too (but we've deprecated the environment variable now), plus a couple more places to look.  

I was tripped up by the complicated sentences and parentheses here and thought a nice compact ordered list would clarify things. I'm OK with just making this suggested change in this section, the others don't seem as complicated.

Reply:
@dbkinder This really opened a can of worms.  In the existing documentation I find four places where the precedence of build variables is specified.  None of them is the same as any other.

I believe in essence option #&#8203;1 expresses "from CMake", which can be any of the three listed sources, of which at least two are probably only consulted when the build area is initialized and between those two I don't know which takes precedence because cmake doesn't document precedence clearly anywhere I can find.

I've switched to your recommended text.